src/librbd/api/Trash.cc

```diff
@@ -26,7 +26,22 @@
 #include "librbd/journal/DisabledPolicy.h"
 #include "librbd/image/ListWatchersRequest.h"
 #include <experimental/map>
-
+#ifdef __GLIBCXX__
+namespace ceph {
+  template<class Map, class Pred>
+  void erase_if(Map& c, Pred pred) {
+    std::experimental::erase_if(c, std::move(pred));
+  }
+}
+#else
+#include <boost/range/algorithm_ext/erase.hpp>
+namespace ceph {
+  template<class Map, class Pred>
+  void erase_if(Map& c, Pred pred) {
+    boost::remove_erase_if(c, std::move(pred));
+  }
+}
+#endif
```

Comment:
If we want to support `libc++`, this should go into a common header.  Alternatively, is there a downside to just using `boost::remove_erase_if` unconditionally?

Reply:
@idryomov turns out `boost::remove_erase_if()` is a very thin wrapper around `std::remove_if()`. and my libc++ tries to move the remaining elements after the erased one by *copying* them. so i created a homebrew version. i'd keep it the a`ceph::erase_if()` right in this file before it has more consumers. also, i don't think we should encourage the usage of C++20 in the C++17 code base, and on the other hand, i don't want to minimize the change to the code for compiling the code with libc++, hence this change.

Reply:
Would it ever get more consumers if it is kept in this file?

Reply:
@idryomov i don't think we should encourage the usage of C++20 in the C++17 code base.

Reply:
+1

If there are problems with `std::experimental::erase_if` why not just remove its usage here, i.e. replace the code with your `std::find_if` version? Or make `ceph::erase_if` common, so it could indeed be reused by other consumers.

Reply:
okay. i will replace it with the `std::find_if()` version.

Reply:
ahh, `std::find_if()` is not good enough in this case,  as there might be multiple `trash_image_specs` which fufill the criteria.